app/models/user.rb

```diff
@@ -5,8 +5,47 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
 
   validates :name, presence: true, length: { maximum: 20 }
-
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships, dependent: :destroy
+  has_many :friends, through: :friendships, dependent: :destroy
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id', dependent: :destroy
+  scope :all_except, ->(user) { where.not(id: user) }
+
+  def friends
+    friendships.map { |friendship| friendship.friend if friendship.status }.compact
+  end
+
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.status }.compact
+  end
```

Comment:
- [ ] Use the same SQL format as in the `confirm_friend` SQL query used for the same reasons as before